ui/src/components/Timeline/util.ts

```diff
@@ -266,3 +266,20 @@ export function useFormValidity(formRef: RefObject<HTMLFormElement>) {
 
   return [isValid, onInput] as const;
 }
+
+export function updateVertex(layout: Layout, updatedVertex: Vertex): Layout {
+  const { vertices } = layout;
+  const index = layout.vertices.findIndex(
+    (vertex) => vertex.entityId === updatedVertex.entityId
+  );
+
+  const newVertices = [...vertices];
+
+  if (index < 0) {
+    newVertices.push(updatedVertex);
+  } else {
+    newVertices.splice(index, 1, updatedVertex);
+  }
+
+  return { vertices: newVertices };
+}
```

Comment:
This logic is now needed in two separate places, so I have extracted it from the state reducer into a separate helper function.